### PR TITLE
tests: add missing dialog tests

### PR DESCRIPTION
**Pull request by rwoll**

### CODE REVIEW

File: tests/test_dialog.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+# //     const dialog = await page.waitForEvent('dialog');
+# //     expect(messages.join()).toContain('confirm "0123456789012345678901234567890123456789012345678…" was shown');
+# //     await dialog.accept('123');
+# //     await promise;
+# //     expect(messages.join()).toContain('confirm "0123456789012345678901234567890123456789012345678…" was accepted');
+# //     await context.close();
+# //   });
+
+# TODO
+# //   it.fail(WEBKIT)('should be able to close context with open alert', async({browser}) => {
```
Comment (rwoll) [inline tests/test_dialog.py:105]: Adding this momentarily

File: tests/test_dialog.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    await on_dialog_actions_complete
+
+
+async def test_should_dismiss_the_confirm_prompt(page: Page, server):
+    on_dialog_actions_complete = asyncio.Future()
+
+    async def on_dialog(dialog: Dialog):
+        on_dialog_actions_complete.set_result(await dialog.dismiss())
+
+    page.on("dialog", lambda dialog: asyncio.create_task(on_dialog(dialog)))
```
Comment (mxschmitt) [inline tests/test_dialog.py:80]:
```suggestion
page.on("dialog", lambda dialog: asyncio.create_task(dialog.dismiss()))
```
one liners are cleaner for that, so you don't need a custom function.

**Comment by pavelfeldman** [comment]
Looks great!!!

**Comment by rwoll** [comment]
Rebased to pickup the test directory change move. All else remains the same.

File: tests/test_dialog.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    async def on_dialog(dialog: Dialog):
+        assert dialog.type == "alert"
+        assert dialog.defaultValue == ""
+        assert dialog.message == "yo"
+        on_dialog_actions_complete.set_result(await dialog.accept())
+
+    page.on("dialog", lambda dialog: asyncio.create_task(on_dialog(dialog)))
+    await page.evaluate("alert('yo')")
+    await on_dialog_actions_complete
```
[the remark this answers is not in the document]
Comment (rwoll) [reply tests/test_dialog.py]: Thanks! I'll follow up

**Pull request merged**